src/states/multilinemps.jl

```diff
@@ -101,7 +101,7 @@ Base.convert(::Type{InfiniteMPS}, st::MultilineMPS) = only(st)
 Base.eltype(t::MultilineMPS) = eltype(t[1])
 Base.copy!(ψ::MultilineMPS, ϕ::MultilineMPS) = (copy!.(parent(ψ), parent(ϕ)); ψ)
 
-Base.isfinite(ψ::MultilineMPS) = false
+Base.isfinite(::Type{<:MultilineMPS}) = false
```

Comment:
Here you probably also need the `isfinite(x) = isfinite(typeof(x))`, since `MultilineMPS` is not an `AbstractMPS`.